webapp/src/components/WebSocketContext.tsx

```diff
@@ -0,0 +1,17 @@
+import { io, Socket } from 'socket.io-client';
+import React, { createContext, PropsWithChildren } from 'react';
+
+const WebSocketContext = createContext<Socket | null>(null);
+let websocket: Socket | null = null;
```

Comment:
Each time `WebSocketProvider` is mounted, it should probably have its own separate instance of `io` that it disconnects from when the component unmounts (probably using [`useEffect`](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1)).

Reply:
@davidvandusen When we were walking through this with the group last week, you mentioned that some parts of this ticket (#174) can be extracted for future tickets. Is this still the case, or should I aim to fulfill the following two things mentioned in the original issue? 
>  The Socket.IO connection has access to the Express session and is able to access the principal_id of the connected user. Users that are not logged in cannot establish a WebSocket connection.

>  Express route handler have access to the Socket.IO instance so that they can emit messages to rooms.

Reply:
This is part of the front-end that @BRashad has taken over to do as part of his ticket #176...so should I leave this alone?

Reply:
I think that is what we agreed on. i'm gonna be working on it.